src/Runner.Listener/SelfUpdater.cs

```diff
@@ -211,6 +301,74 @@ private async Task DownloadLatestRunner(CancellationToken token)
                 }
             }
 
+            var trimmedPackageRestoreTasks = new List<Task<bool>>();
+            if (!fallbackToFullPackage)
```

Comment:
Should we consider making this a function, and just calling recursively if it fails and pass this value in? May cleanup some of the dupe create/delete code